### PR TITLE
refactor(group): migrate AddGroupUsers handler to membership

### PR DESCRIPTION
**Pull request by whoAbhishekSah**
## Summary

Migrates the legacy `AddGroupUsers` RPC handler off `groupService.AddUsers` and onto the membership package. The proto RPC itself is unchanged — this is a pure internal refactor that completes the migration so every group-membership write in the codebase flows through `membership.SetGroupMemberRole`.

With this PR, `core/group/service.go` is out of the membership-write business: `AddMember`, `AddUsers`, and `addMemberPolicy` are removed, and the `RelationService` / `PolicyService` interfaces on the group service are narrowed to the methods still in use.

This PR also collapses `membership.AddGroupMember` into `SetGroupMemberRole`. After the upsert relaxation, both methods did the same work — the three internal callers (`OnGroupCreated`, `AddGroupUsers` handler, `invitation.Accept`) are all fresh-add sites and either guaranteed-new or already pre-filtered, so the strict `ErrAlreadyMember` signal isn't needed. The "added" vs "role_changed" audit distinction is preserved by `SetGroupMemberRole`'s own branching.

Finally, applies the same atomic last-owner guard introduced for org in #1590 to the group path: `validateMinGroupOwnerConstraint` now returns the resolved owner role ID, which `SetGroupMemberRole` forwards to `replacePolicy` so the owner-role delete uses `DeleteWithMinRoleGuard` (atomic in the DB) instead of a plain delete after a stale read. `ErrLastOwnerRole` from the generic guard is converted to `ErrLastGroupOwnerRole` at the call site so the user-facing message stays group-specific.

## Behavior notes

- `AddGroupUsers` now goes through the org-membership pre-check on the upsert-add path. A freshly-created user who is not yet an org member will be rejected with `ErrNotOrgMember`. The 4 e2e regression tests that previously relied on adding non-org-users to a group have been updated to add the user to the org first.
- `AddGroupUsers` callers (existing SDK) are otherwise unaffected — same proto, same response shape.

## Test plan

- [x] `go build ./...`
- [x] `go test ./core/group/... ./core/invitation/... ./core/membership/... ./internal/api/v1beta1connect/...`
- [x] `gofmt -l` clean
- [x] New unit test covers the TOCTOU race path on the group last-owner guard
- E2E regression and integration tests run in CI

### CODE REVIEW

**Comment by whoAbhishekSah** [comment]
## Manual Testing Results ✅

Tested all edge cases for `AddGroupUsers` migration to `SetGroupMemberRole` and the atomic last-owner guard.

### AddGroupUsers

| # | Test Case | Expected | Result |
|---|-----------|----------|--------|
| 1 | Add org member to group | SUCCESS | ✅ |
| 2 | Add multiple users at once | SUCCESS | ✅ |
| 3 | Verify members added | 4 users | ✅ |
| 4 | Idempotent add (re-add same user) | SUCCESS, count unchanged | ✅ |
| 5 | Add non-org-member | `failed_precondition: principal is not a member of the organization` | ✅ |
| 6 | Add non-existent user | `not_found: user doesn't exist` | ✅ |
| 7 | Add to non-existent group | `permission_denied` | ✅ |

### SetGroupMemberRole

| # | Test Case | Expected | Result |
|---|-----------|----------|--------|
| 8 | Promote member to owner | SUCCESS | ✅ |
| 9 | Demote owner (another owner exists) | SUCCESS | ✅ |
| 10 | Demote last owner | `failed_precondition: group must have at least one owner...` | ✅ |
| 11 | Add non-org-member | `failed_precondition: principal is not a member of the organization` | ✅ |
| 12 | Promote second owner | SUCCESS | ✅ |
| 13 | Demote first owner (second exists) | SUCCESS | ✅ |
| 14 | Use invalid role (org role on group) | `invalid_argument: role is not valid for group scope` | ✅ |
| 15 | Invalid principal type | `invalid_argument: unsupported principal type` | ✅ |
| 16 | Idempotent role set | SUCCESS | ✅ |

### RemoveGroupUser

| # | Test Case | Expected | Result |
|---|-----------|----------|--------|
| 17 | Remove user from group | SUCCESS | ✅ |
| 18 | Verify user removed | User not in list | ✅ |
| 19 | Remove last owner | `invalid_argument: group must have at least one owner...` | ✅ |

### Group Independence & Cross-org Security

| # | Test Case | Expected | Result |
|---|-----------|----------|--------|
| 20 | Add users to second group | SUCCESS | ✅ |
| 21 | Groups are independent | Different member lists | ✅ |
| 22 | Cross-org: SetGroupMemberRole | `failed_precondition: principal is not a member of the organization` | ✅ |
| 23 | Cross-org: AddGroupUsers | `failed_precondition: principal is not a member of the organization` | ✅ |

### Notes

- TOCTOU race condition for last-owner guard is covered by unit test in `core/membership/service_test.go`
- All error messages are now consistent between `AddGroupUsers` and `SetGroupMemberRole`

File: internal/api/v1beta1connect/group.go [hunk 1]
```diff
@@ -409,11 +409,34 @@ func (h *ConnectHandler) AddGroupUsers(ctx context.Context, request *connect.Req
 		}
 	}
 
-	if err := h.groupService.AddUsers(ctx, request.Msg.GetId(), request.Msg.GetUserIds()); err != nil {
-		errorLogger.LogServiceError(ctx, request, "AddGroupUsers.AddUsers", err,
+	var joinedErr error
+	for _, userID := range request.Msg.GetUserIds() {
+		if err := h.membershipService.SetGroupMemberRole(ctx, request.Msg.GetId(), userID, schema.UserPrincipal, schema.GroupMemberRole); err != nil {
+			joinedErr = errors.Join(joinedErr, err)
+		}
```
[the remark this answers is not in the document]
Comment (whoAbhishekSah) [reply internal/api/v1beta1connect/group.go:416]: Yeah, it is acceptable.

File: core/invitation/service.go [hunk 2]
```diff
@@ -339,10 +339,7 @@ func (s Service) Accept(ctx context.Context, id uuid.UUID) error {
 				}
 			}
 			if !alreadyGroupMember {
-				if err = s.groupSvc.AddMember(ctx, grp.ID, authenticate.Principal{
-					ID:   userOb.ID,
-					Type: schema.UserPrincipal,
-				}); err != nil {
+				if err = s.membershipSvc.AddGroupMember(ctx, grp.ID, userOb.ID, schema.UserPrincipal, schema.GroupMemberRole); err != nil {
```
[the remark this answers is not in the document]
Comment (rohilsurana) [reply core/invitation/service.go]: worth checking

**Review by rohilsurana — APPROVED** [review]
